qemu/tests/block_iscsi_fault_disk.py

```diff
@@ -0,0 +1,159 @@
+        online_cmd += " && echo clean >> " + disk
+        online_cmd += " && echo attributes disk clear readonly >> " + disk
+        online_cmd += " && echo detail disk >> " + disk
+        online_cmd += " && diskpart /s " + disk
+        online_cmd += " && del /f " + disk
+        return session.cmd(online_cmd % index, timeout=timeout)
+
+    def _get_window_disk_index_by_uid(wwn):
+        cmd = "powershell -command \"get-disk|?"
+        cmd += " {$_.UniqueId -eq '%s'}|select number|FL\"" % wwn
+        status, output = session.cmd_status_output(cmd)
+        if status != 0:
+            test.fail("execute command fail: %s" % output)
+        logging.debug(output)
+        output = "".join([s for s in output.splitlines(True) if s.strip()])
+
+        info = output.split(":")
+        if len(info) > 1:
+            return info[1].strip()
+
+        cmd = "powershell -command \"get-disk| FL\""
+        output = session.cmd_output(cmd)
+        logging.debug(output)
+        test.fail("Not find expected disk:" + wwn)
+
+    def _get_disk_wwn(devname):
+        cmd = "lsblk -ndo WWN " + devname
+        output = process.system_output(cmd, shell=True).decode()
+        wwn = output.replace("0x", "")
+        return wwn
+
+    vm = None
+    iscsi = None
+    params['scsi_debug_disk'] = None
+    params['mapper_disk'] = None
+    timeout = float(params.get("timeout", 360))
+    expected_timeout = int(params.get("expected_timeout"))
+    os_type = params["os_type"]
+    guest_cmd = params["guest_cmd"]
+    host_kill_command = params["host_kill_command"]
+
+    try:
+        logging.info("Prepare fault disk.")
+        _prepare_fault_disk()
+        logging.info("Create iscsi disk disk.")
+        base_dir = data_dir.get_data_dir()
+        iscsi = Iscsi.create_iSCSI(params, base_dir)
+        iscsi.login()
+
+        dev_name = utils_misc.wait_for(lambda: iscsi.get_device_name(), 60)
+        if not dev_name:
+            test.error('Can not get the iSCSI device.')
+
+        logging.info('Create host disk %s', dev_name)
+        disk_wwn = _get_disk_wwn(dev_name)
+        params["image_name_stg0"] = dev_name
+
+        logging.info('Booting vm...')
+        params['start_vm'] = 'yes'
+        vm = env.get_vm(params['main_vm'])
+        env_process.process(test, params, env, env_process.preprocess_image,
+                            env_process.preprocess_vm)
+        session = vm.wait_for_login(timeout=600)
+
+        if os_type == 'windows':
+            guest_cmd = utils_misc.set_winutils_letter(session, guest_cmd)
+            disk_drive = _get_window_disk_index_by_uid(disk_wwn)
+            _online_disk_windows(disk_drive)
+        else:
+            disk_drive = get_linux_drive_path(session, disk_wwn)
+
+        guest_cmd = guest_cmd % disk_drive
+        logging.debug("guest_cmd:%s", guest_cmd)
+
+        logging.info("Execute io in guest...")
+        session.sendline(guest_cmd)
+        time.sleep(10)
```

Comment:
Shall we add a step to make sure the IO process is still running before killing the VM process?
Checked the corresponding manual case:
```
Step:
2. do io in guest
dd if=/dev/zero of=/dev/sda bs=1M count=100 oflag=direct
Expected Result:
the io operation not finished in short time due to low response on storage server
```
So I think we need to add a check point that judge whether `the io operation not finished in short time` after run the IO command.

Reply:
The io must be blocked due to falut disk.
If the IO  completed,The qemu process will be truly killed immediately result in timeout check exception.

Reply:
Yes, io must be blocked, but I think it is expected without any product bug.
In case there is an issue that the io operation is finished in short time due to some product bug, so we will miss it.
Anyway, that's just my consider the risk, but the actual action depend you, it is also okay for me which action you take.

Reply:
The risk has been covered:

if kill_min_timeout < real_timeout < kill_max_timeout:
            logging.info("Succeed kill timeout: %d", real_timeout)
else:
            test.fail("Kill timeout %d not in range (%d , %d)" % (
                real_timeout, kill_min_timeout, kill_max_timeout))

Reply:
Okay, got it, looks good to me.